pkg/plugin/repository/GlobalPluginRepository.go

```diff
@@ -181,10 +197,13 @@ type GlobalPluginRepositoryImpl struct {
 	dbConnection *pg.DB
 }
 
-func (impl *GlobalPluginRepositoryImpl) GetMetaDataForAllPlugins() ([]*PluginMetadata, error) {
+func (impl *GlobalPluginRepositoryImpl) GetMetaDataForAllPlugins(stageType StageType) ([]*PluginMetadata, error) {
 	var plugins []*PluginMetadata
 	err := impl.dbConnection.Model(&plugins).
-		Where("deleted = ?", false).Select()
+		Join("INNER JOIN plugin_stage_mapping psm on psm.plugin_id=plugin_metadata.id").
+		Where("plugin_metadata.deleted = ?", false).
+		Where("psm.stage_type= 2 or psm.stage_type= ?", stageType).
```

Comment:
Please check for Grouping here by verifying the query

Reply:
already verified